docs/sources/operators-guide/deploying-grafana-mimir/jsonnet/configuring-autoscaling.md

```diff
@@ -0,0 +1,100 @@
+
+HPA is not configured directly in Jsonnet but it's created and updated by KEDA.
+KEDA is an operator, running in the Kubernetes cluster, which is responsible to simplify the setup of HPA with custom metrics (Prometheus in our case).
+
+## How KEDA works
+
+KEDA is a Kubernetes operator aiming to simplify the wiring between HPA and Prometheus.
+
+Kubernetes HPA, out of the box, is not capable of autoscaling based on metrics scraped by Prometheus, but it allows to configure a custom metrics API server which proxies metrics from a datasource (e.g. Prometheus) to Kubernetes.
+Setting up the custom metrics API server for Prometheus in a Kubernetes can be a tedious operation, so KEDA offers an operator to set it up automatically.
+KEDA supports proxying metrics for a variety of sources, including Prometheus.
+
+### KEDA in a nutshell
+
+- Runs an operator and an external metrics server.
+- The metrics server supports proxying for many metric sources, including Prometheus.
+- The operator watches for `ScaledObject` custom resource definition (CRD), defining the minimum and maximum replicas, and scaling trigger metrics of a Deployment or StatefulSet, and then configures the related HPA resource. You don't create the HPA resource in Kubernetes, but the operator creates it for you whenever a `ScaledObject` CRD is created (and keeps it updated for its whole lifecycle).
+
+Refers to [KEDA documentation](https://keda.sh) for more information.
+
+### What happens if KEDA is unhealthy
+
+The autoscaling of deployments is always managed by HPA, which is a native Kubernetes feature.
+KEDA, as we use it, never changes the number of replicas of Mimir Deployments or StatefulSets.
+
+However, if KEDA is not running successfully, there are consequences for Mimir autoscaling too:
+
+- `keda-operator` is down (not critical): changes to `ScaledObject` CRD will not be reflected to the HPA until the operator will get back online. HPA functionality is not affected.
+- `keda-operator-metrics-apiserver` is down (critical): HPA is not able to fetch updated metrics and it will stop scaling the deployment until metrics will be back. The deployment (e.g. queriers) will keep working but, in case of any surge of traffic, HPA will not be able to detect it (because of a lack of metrics) and so will not scale up.
+
+The [alert `MimirQuerierAutoscalerNotActive`]({{< relref "../../visualizing-metrics/_index.md" >}}) fires if HPA is unable to scale the deployment for any reason (e.g. unable to scrape metrics from KEDA metrics API server).
+
+## How Kubernetes HPA works
+
+Refer to Kubernetes [Horizontal Pod Autoscaling](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) documentation to have a full understanding of how HPA works.
+
+## How to enable autoscaling
+
+The following Jsonnet configuration snippet shows an example of how to enable Mimir autoscaling with Jsonnet:
```

Comment:
It is unclear for me here whether KEDA will be installed by just running this, or I need to install KEDA separately. Can we make that explicit?

Reply:
Great, thanks.